gems/smithy-client/lib/smithy-client/base.rb

```diff
@@ -23,10 +23,11 @@ def initialize(plugins, options)
         after_initialize(plugins)
       end
 
-      # @return [Configuration<Struct>]
+      # @return [Struct]
       attr_reader :config
 
       # @return [HandlerList]
+      # @api private
```

Comment:
I think this is and should be public.

Reply:
I'm not really sure we want to expose the handler list as public - I'd generally prefer to keep the public surface area as small as possible to start and if theres a reason or if its useful later, we can open it up.

Reply:
I think it's public to anyone writing plugins.

Reply:
As far as I can tell, only the Configuration::DefaultResolver is actually exposed when developing plugins (through the default block on an option), but I'll remove the api private from it.